File: infosupport/infoserver/src/main/java/app/models/Patient.java
package app.models;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Entity(name = "Patient")
public class Patient implements Serializable {

  @Id
  @Column(name = "user_id")
  private Integer user_id;

  @Column(name = "dateOfBirth")
  private LocalDate dateOfBirth;

  @Column(name = "gender")
  private String gender;

  @Column(name = "allergies")
  private String allergies;

  @Column(name = "gp_user_id")
  private int gp_user_id;
  @Column(name = "email")
  private String email;

  @Column(name = "firstname")
  private String firstname;

  @Column(name = "lastname")
  private String lastname;

  @Column(name = "password")
  private String password;

  @Column(name = "phonenumber")
  private String phonenumber;

  public Patient(){ }

  public Integer getUser_id() {
    return user_id;
  }

  public void setUser_id(Integer user_id) {
    this.user_id = user_id;
  }

  public LocalDate getDateOfBirth() {
    return dateOfBirth;
  }

  public void setDateOfBirth(LocalDate dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
  }

  public String getGender() {
    return gender;
  }

  public void setGender(String gender) {
    this.gender = gender;
  }

  public String getAllergies() {
    return allergies;
  }

  public void setAllergies(String allergies) {
    this.allergies = allergies;
  }

  public int getGp_user_id() {
    return gp_user_id;
  }

  public void setGp_user_id(int gp_user_id) {
    this.gp_user_id = gp_user_id;
  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public String getFirstname() {
    return firstname;
  }

  public void setFirstname(String firstname) {
    this.firstname = firstname;
  }

  public String getLastname() {
    return lastname;
  }

  public void setLastname(String lastname) {
    this.lastname = lastname;
  }

  public String getPassword() {
    return password;
  }

  public void setPassword(String password) {
    this.password = password;
  }

  public String getPhonenumber() {
    return phonenumber;
  }

  public void setPhonenumber(String phonenumber) {
    this.phonenumber = phonenumber;
  }
}

File: infosupport/src/app/models/patient.ts
import {User} from "./user";

export class Patient {

  private _user_id: number;
  private _dateOfBirth: Date;
  private _gender: string;
  private _allergies: string;
  private _email: string;
  private _firstname: string;
  private _lastname: string;
  private _phonenumber: string;
  private _password: string;


  constructor(user_id?: number, dateOfBirth?: Date, gender?: string, allergies?: string, email?: string, firstname?: string, lastname?: string, phonenumber?: string, password?: string) {
    this._user_id = user_id;
    this._dateOfBirth = dateOfBirth;
    this._gender = gender;
    this._allergies = allergies;
    this._email = email;
    this._firstname = firstname;
    this._lastname = lastname;
    this._phonenumber = phonenumber;
    this._password = password;
  }

  get user_id(): number {
    return this._user_id;
  }

  set user_id(value: number) {
    this._user_id = value;
  }

  get dateOfBirth(): Date {
    return this._dateOfBirth;
  }

  set dateOfBirth(value: Date) {
    this._dateOfBirth = value;
  }

  get gender(): string {
    return this._gender;
  }

  set gender(value: string) {
    this._gender = value;
  }

  get allergies(): string {
    return this._allergies;
  }

  set allergies(value: string) {
    this._allergies = value;
  }


  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  get firstname(): string {
    return this._firstname;
  }

  set firstname(value: string) {
    this._firstname = value;
  }

  get lastname(): string {
    return this._lastname;
  }

  set lastname(value: string) {
    this._lastname = value;
  }

  get phonenumber(): string {
    return this._phonenumber;
  }

  set phonenumber(value: string) {
    this._phonenumber = value;
  }

  get password(): string {
    return this._password;
  }

  set password(value: string) {
    this._password = value;
  }
}
